<a href="https://colab.research.google.com/github/PosgradoMNA/actividades-de-aprendizaje-jesusrincon1328/blob/main/Data%20Analysis%20with%20Python/Module_4_Model_Development.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
path = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DA0101EN-SkillsNetwork/labs/Data%20files/automobileEDA.csv"

In [3]:
df = pd.read_csv(path)
df.head()

symboling  normalized-losses         make aspiration num-of-doors  \
0          3                122  alfa-romero        std          two   
1          3                122  alfa-romero        std          two   
2          1                122  alfa-romero        std          two   
3          2                164         audi        std         four   
4          2                164         audi        std         four   

    body-style drive-wheels engine-location  wheel-base    length  ...  \
0  convertible          rwd           front        88.6  0.811148  ...   
1  convertible          rwd           front        88.6  0.811148  ...   
2    hatchback          rwd           front        94.5  0.822681  ...   
3        sedan          fwd           front        99.8  0.848630  ...   
4        sedan          4wd           front        99.4  0.848630  ...   

   compression-ratio  horsepower  peak-rpm city-mpg highway-mpg    price  \
0                9.0       111.0    5000.0       21          27  13495.0   
1                9.0       111.0    5000.0       21          27  16500.0   
2                9.0       154.0    5000.0       19          26  16500.0   
3               10.0       102.0    5500.0       24          30  13950.0   
4                8.0       115.0    5500.0       18          22  17450.0   

  city-L/100km  horsepower-binned  diesel  gas  
0    11.190476             Medium       0    1  
1    11.190476             Medium       0    1  
2    12.368421             Medium       0    1  
3     9.791667             Medium       0    1  
4    13.055556             Medium       0    1  

[5 rows x 29 columns]

<h2>1. Linear Regression and Multiple Linear Regression</h2>

**Simple Linear Regression**

Es un metodo que nos ayuda a entender la relación entre dos variables


*   El predictor/variable independiente(X)
*   La respuesta/variable dependiente(lo que queremos predecir)(Y)







In [4]:
from sklearn.linear_model import LinearRegression

In [8]:
lm = LinearRegression()
lm

LinearRegression()

In [9]:
X = df[['highway-mpg']]
Y = df['price']

In [10]:
lm.fit(X,Y)

LinearRegression()

In [11]:
Yhat=lm.predict(X)
Yhat[0:5]  

array([16236.50464347, 16236.50464347, 17058.23802179, 13771.3045085 ,
       20345.17153508])

In [12]:
lm.intercept_

38423.3058581574

In [13]:
lm.coef_

array([-821.73337832])

Cual es el modelo lineal estimado?

Debemos tener un modelo lineal final con la siguiente estructura:

yhat = a + bX

por lo que tendriamos:

**Price** = 38423.31 - 821.73 x **highway-mpg**

In [15]:
#Crear un objeto de regresion lineal llamado lm1

lm1 = LinearRegression()
lm1

LinearRegression()

In [16]:
#Entrena el modelo utilizando "engine-size" como variable independiente y "price " como variable dependiente.

lm1.fit(df[["engine-size"]], df[["price"]])
lm1

LinearRegression()

In [17]:
#Slope
lm1.coef_


array([[166.86001569]])

In [19]:
#Intercept
lm1.intercept_

array([-7963.33890628])

In [21]:
#equiación predictiva lineal


Price=-7963.34 + 166.86*df['engine-size']